src/java/org/apache/cassandra/schema/ColumnMetadata.java

```diff
@@ -320,7 +387,8 @@ private boolean equalsWithoutType(ColumnMetadata other)
             && position == other.position
             && ksName.equals(other.ksName)
             && cfName.equals(other.cfName)
-            && Objects.equals(mask, other.mask);
+            && Objects.equals(mask, other.mask)
+            && Objects.equals(columnConstraints, other.columnConstraints);
```

Comment:
Just to confirm, do we want to explicitly only check memory address equality between columnConstraints instances rather than contents therein? Since we're logically allowing non-memory identical comparisons with other fields including relying on the `.equals` implementation within `ColumnMask`. Just curious what the thinking was here.

Reply:
I overrode the equals on the relevant classes

Reply:
Looks like it's still missing in `FunctionColumnConstraint.java`.